symengine/monomials.cpp

```diff
@@ -8,6 +8,7 @@ namespace SymEngine {
 void monomial_mul(const vec_int &A, const vec_int &B, vec_int &C)
 {
     size_t n = A.size();
+    C.resize(n);
```

Comment:
You don't want to be doing resize at this place, since this would slow things down.

Can you write a test case that shows the failure? Then we'll see how to best fix it.